src/TimeModeling/Types/abstract.jl

```diff
@@ -0,0 +1,182 @@
+export subsample
+
+############################################################################################################################
+# Base multi source abstract type
+abstract type judiMultiSourceVector{T} <: DenseVector{T} end
```

Comment:
An OOC judiVector is not DenseVector because it is not stored in contiguous memory?

Reply:
It is dense because you never do OOC computation and always reads and load the data into a dense vector to compute anything.